src/libgit2/transports/smart_protocol.c

```diff
@@ -317,7 +322,26 @@ static int wait_while_ack(gitno_buffer *buf)
 	return 0;
 }
 
-int git_smart__negotiate_fetch(git_transport *transport, git_repository *repo, const git_remote_head * const *wants, size_t count)
+static int cap_not_sup_err(const char *cap_name)
+{
+	git_error_set(GIT_ERROR_NET, "server doesn't support %s", cap_name);
+	return GIT_EINVALID;
+}
+
+/* Disables server capabilities we're not interested in */
+static int setup_caps(transport_smart_caps *caps, const git_fetch_negotiation *wants)
+{
+	if (wants->depth) {
+		if (!caps->shallow)
```

Comment:
Is `wants` specified the same as the fetch options? If so, the default is `depth == -1`, which coerces to boolean `1` in C. If that's the case, then...

```suggestion
	if (wants->depth > 0) {
		if (!caps->shallow)
```